src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder(
+                "Report for period " + dateFrom + " - " + dateTo);
+        int space = 1;
+        LocalDate localDateFrom = LocalDate.of(
+                Integer.parseInt(dateFrom.substring(6, 10)),
+                Integer.parseInt(dateFrom.substring(3, 5)),
+                Integer.parseInt(dateFrom.substring(0, 2))
+        );
+        LocalDate localDateTo = LocalDate.of(
+                Integer.parseInt(dateTo.substring(6, 10)),
+                Integer.parseInt(dateTo.substring(3, 5)),
+                Integer.parseInt(dateTo.substring(0, 2))
+        );
+        for (String name : names) {
+            int salary = 0;
+            for (String oneData : data) {
+                String[] oneDataSplit = oneData.split(" ");
+                LocalDate localOneData = LocalDate.of(
```

Comment:
As mentioned above, you should use `DateTimeFormatter` class to parse "yyy-MM-dd" formatted string into LocalDate - no need to call for substrings separately